src/redux/actionCreators.ts

```diff
@@ -0,0 +1,1474 @@
+import * as isEqual from 'fast-deep-equal';
```

Comment:
herein lies everything we were previously doing in `ComponentMap` and `FlowMutator`. the `redux-thunk` middleware passes us `redux`'s `getState` method, which allows us to read from state at any point. action creators like `addNode` update our flow definition so we need to call `getState` right after to get that updated definition if we want to read from it right after we add a node (some examples of that below). i imagine we can optimize these for purity as we go (probably not great to be calling `getState` everywhere).